### PR TITLE
Dudedev 28117 add escaping information to usage text for security utility

### PR DESCRIPTION
**Pull request by dudedev**
- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [ ] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [ ] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

################################################################################################

It completes this task. https://github.com/OpenLiberty/open-liberty/issues/28117

For full details, please see this wiki page: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions
################################################################################################

### CODE REVIEW

**Comment by dudedev** [comment]
#build

**Comment by una-tapa** [comment]
#build

File: dev/com.ibm.ws.security.utility/resources/com/ibm/ws/security/utility/resources/UtilityOptions.nlsprops [hunk 1]
```diff
@@ -53,7 +53,11 @@ help.usage.options=\
 #------------------------------\n at 72 chars Leading "\ \ \ \ "-------\n\#
 #------------------------------\n at 72 chars Leading "\t"-------------\n\#
 encode.desc=\
-\tEncode the provided text.
+\tEncode the provided text.\n\n\
+\tIf your password includes special characters, you must escape each special character to help ensure that the password is properly encoded.\n\
+\tSpecial characters and escape characters might vary according to your operating system.\n\
+\tFor example, on Unix systems, pa$$W0rd must be provided as pa\\$\\$W0rd.\n\
```
Comment (helyarp) [inline dev/com.ibm.ws.security.utility/resources/com/ibm/ws/security/utility/resources/UtilityOptions.nlsprops]: Replace "Unix" with "UNIX"

**Review by helyarp — COMMENTED** [review]
Suggested change

File: README.md [hunk 2]
```diff
@@ -74,9 +74,13 @@ Our [CONTRIBUTING](https://github.com/OpenLiberty/open-liberty/blob/release/CONT
 
 2. Run a gradle build.
 
+   Pre-requisite - Environment variable `JAVA_21_HOME` must be set.
```
Comment (jhanders34) [inline README.md]:
```suggestion
   Prerequisite - Environment variable `JAVA_HOME` must point to a Java 17 or Java 21 SDK.  If setting `JAVA_HOME` to  Java 17, you will also need to set `JAVA_21_HOME` to a Java 21 SDK.
```

File: README.md [hunk 3]
```diff
@@ -74,9 +74,13 @@ Our [CONTRIBUTING](https://github.com/OpenLiberty/open-liberty/blob/release/CONT
...
+
         cd open-liberty/dev
         ./gradlew cnf:initialize
         ./gradlew assemble
+
+    Troubleshoot - If `./gradlew assemble` shows Out of Memory error then you could increase the memory in `dev/gradle.properties` file. Stop gradle daemons after you change the memory using `./gradlew --stop`. If it does not help, then try cleaning up maven repository (using `rm -rf ~/.m2`) and gradle caches (using `rm -rf ~/.gradle/caches`).
```
Comment (jhanders34) [inline README.md]:
```suggestion
    Troubleshooting - If `./gradlew assemble` shows out of memory error, you can increase the memory in `dev/gradle.properties` file. Stop the gradle daemon after you change the memory setting by using `./gradlew --stop`. If it does not help the problem, try cleaning up your maven repository cache (using `rm -rf ~/.m2`) and gradle caches (using `rm -rf ~/.gradle/caches`).
```

File: dev/com.ibm.ws.security.utility/resources/com/ibm/ws/security/utility/resources/UtilityOptions.nlsprops (hunk 1 above)
Comment (dudedev) [reply dev/com.ibm.ws.security.utility/resources/com/ibm/ws/security/utility/resources/UtilityOptions.nlsprops]: done

File: README.md (hunk 2 above)
Comment (dudedev) [reply README.md]: Updated as suggested.

File: README.md (hunk 3 above)
Comment (dudedev) [reply README.md]: Updated as suggested.

**Comment by dudedev** [comment]
> ### Code analysis and actions
> DO NOT DELETE THIS COMMENT.
> * 1 product code files were changed.
> * **Please describe in a separate comment how you tested your changes.**
> * 1 messages files were changed and need an L2 review.
> * @OpenLiberty/message-reviewer Please review.
> * dev/com.ibm.ws.security.utility/resources/com/ibm/ws/security/utility/resources/UtilityOptions.nlsprops
> * 1 NLS files were changed and need an ID review.
> * @OpenLiberty/message-reviewer Please review.
> * dev/com.ibm.ws.security.utility/resources/com/ibm/ws/security/utility/resources/UtilityOptions.nlsprops

> How I tested the code?

After code change, I ran the build and executed following command to get the help text.
`./securityUtility encode --help`
This resulted in following output.
```
Usage:
        securityUtility encode [options]

Description:
        Encode the provided text.

        If your password includes special characters, you must escape each special character to help ensure that the password is properly encoded.
        Special characters and escape characters might vary according to your operating system.
        For example, on UNIX systems, pa$$W0rd must be provided as pa\$\$W0rd.
        For more information, see the Open Liberty securityUtility encode documentation.


Options:
    --encoding=[xor|aes|hash]
        Specify how to encode the password. Supported encodings are xor, aes,
        and hash. The default encoding is xor. 

    --key=[key]
        Specify a key to be used when encoding using AES. This string will be
        hashed to produce an encryption key which will be used to encrypt and
        decrypt the password. The key can be provided to the server by
        defining the variable wlp.password.encryption.key whose value is the
        key. If this option is not provided a default key shall be used.

    --keyLabel=[keyLabel]
        Specify the label of the SAF certificate of the AES encryption key.

    --keyring=[keyring]
        Specify the SAF keyring where the AES encryption key is located.

    --keyringType=[keyringType]
        Specify the SAF keyring type of the keyring where the AES encryption
        is located.

    --listCustom
        Display the information of the custom password encryption in
        JavaScript Object Notation (JSON) format.
        The information consists of:
        name : the custom password encryption algorithm name
        featurename : the feature name
        description : the description of the custom password encryption

    --notrim
        Specify whether space characters are removed from the beginning and
        end of the specified text. If this option is specified, the provided
        text will be encoded as it is.
        If this option is not specified, space characters from the beginning
        and end of the specified text will be removed. 

    textToEncode
        If no arguments are specified, the tool will enter the interactive
        mode; otherwise, the provided text will be encoded.
        Text with spaces must be fully quoted if specified as an argument.
```

**Comment by dudedev** [comment]
> I cannot process this build request due to one or both of the following:
> 
> * The comment author is not a member of the organization.
> * The pull request creator did not send a signed CLA.

I have already signed the ICLA and it has been processed.

**Review by helyarp — COMMENTED** [review]
Looks good!

**Comment by dudedev** [comment]
@helyarp @una-tapa I have updated the copyright year. Please review.

File: dev/com.ibm.ws.security.utility/resources/com/ibm/ws/security/utility/resources/UtilityOptions.nlsprops [hunk 4]
```diff
@@ -1,5 +1,5 @@
 ###############################################################################
-# Copyright (c) 2010, 2020 IBM Corporation and others.
+# Copyright (c) 2010, 2020, 2024 IBM Corporation and others.
```
Comment (helyarp) [inline dev/com.ibm.ws.security.utility/resources/com/ibm/ws/security/utility/resources/UtilityOptions.nlsprops]: Change the copyright to show _first_year_, _last_year_ (`2010, 2024`), thus remove `2020, `

**Review by helyarp — COMMENTED** [review]
Requested change

File: dev/com.ibm.ws.security.utility/resources/com/ibm/ws/security/utility/resources/UtilityOptions.nlsprops (hunk 4 above)
Comment (dudedev) [reply dev/com.ibm.ws.security.utility/resources/com/ibm/ws/security/utility/resources/UtilityOptions.nlsprops]: removed 2020

**Review by helyarp — COMMENTED** [review]
Looks good!

**Review by una-tapa — APPROVED** [review]
LGTM

**Comment by una-tapa** [comment]
#libby

**Comment by jhanders34** [comment]
!libby

**Pull request merged**